pkg/operator/operator.go

```diff
@@ -198,3 +210,52 @@ func (o *Operator) ensureDefaultIngressController() error {
 	log.Info("created default ingresscontroller", "namespace", ic.Namespace, "name", ic.Name)
 	return nil
 }
+
+func ensureServicesHaveIdleAnnotation(cl client.Client) error {
+	endpointsList := &corev1.EndpointsList{}
+	err := cl.List(context.TODO(), endpointsList, &client.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to list endpoints in all namespaces: %v", err)
+	}
+
+	for _, endpoints := range endpointsList.Items {
```

Comment:
If possible, I'd like to get data on how this affects start time on a cluster with, say, a hundred thousand endpoints.  This might best be done asynchronously (i.e., in a goroutine), even if we leave it it as a one-off for the time being.

Reply:
Andy and I thought it would make the most sense for the operator to ensure that the idle annotations exist on services before creating the router deployment, since update the service annotations would not guarantee a router reload necessarily. Doing this asynchronously could make the behavior racey, so im not sure the best path forward. Thoughts?

Reply:
I share this concern. My concern is subsequently getting the idle annotations on a service but no corresponding endpoint changes. It's only in `createRouterEndpoints()` that we check to for _idleness_ and we only get there if there are endpoint additions/deletions. It seems that we could get out of sync if done asynchronously.

Reply:
Keep in mind that this logic is only needed during upgrades, in which case the router deployment already exists.  But I think your reasoning holds if you s/creating the router deployment/updating the router deployment to use the image pullspec in the release payload/.  Based on that reasoning, doing this migration synchronously makes sense.  However, this puts the migration in the upgrade path, and I wonder (a) how much it increases time to upgrade and (b) how long the operator will be out of service and not reconciling its operands.

Reply:
@Miciah I just ran a rough timing estimate here and on a 4.7 cluster with ~25k endpoints, and 10 idled services, the new idle annotation verification function took about 2 seconds to run. Is that a concerning amount of time in your opinion?

Reply:
Note that my tests were performed by running the operator locally, so in-cluster times would be probably be fast given lower API request latency.

Reply:
If it scales linearly (i.e., a cluster with 100k endpoints would take 8 seconds, and 200k endpoints would take 16 seconds), that doesn't seem too onerous.

Reply:
A similar test with 50k endpoints took ~3.8 seconds, so I think we have enough evidence to suggest it scales linearly.